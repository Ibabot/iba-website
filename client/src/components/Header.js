import React from 'react';
import './header.scss';
import logo from '../assets/img/isb-logo.svg';


function Header() {
    return(
        <header className="web-header">
            <div className="logo">
                <img src={logo} alt="Íslandsbankinn logo" className="logo" />
            </div>            
            <input type="text" className="input"></input>
        </header>
    )
}

export default Header;
