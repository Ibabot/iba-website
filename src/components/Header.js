import React from 'react';
import './header.scss';
import logo from '../assets/img/isb-logo.svg';


function Header() {
    return (
      <header className="web-header" id="web-header">
        <div className="logo" data-aos="fade-right">
          <img src={logo} alt="Íslandsbankinn logo" className="logo" />
        </div>
        <div
          className="web-header-text"
          data-aos="fade-left"
          data-aos-delay="100"
        >
          <h1>Segðu halló við Íba</h1>
          <h2 className="bottom-space">
            <strong>Íba er snjallmenni Íslandsbanka.</strong>
          </h2>
          <h2>
            Íba getur svarað almennum spurningum og fyrirspurnum hér í
            spjallinu.
          </h2>
          <a
            data-testid="get-assistance"
            className="btn btn-red"
            href="https://www.islandsbanki.is/is/vara/thjonusta/hafa-samband"
            rel="noopener noreferrer"
            target="_blank"
          >
            Hafðu samband við þjónustuver bankans
          </a>
          <p>
            Ég er líka á{" "}
            <a
              data-testid="open-facebook"
              href="https://www.messenger.com/t/iba.spjallmenni"
              rel="noopener noreferrer"
              target="_blank"
            >
              Facebook
            </a>
          </p>
        </div>
      </header>
    );
}

export default Header;