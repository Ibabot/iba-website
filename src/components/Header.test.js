import React from "react";
import ReactDOM from "react-dom";
import Header from "./Header";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import testRenderer from "react-test-renderer";

afterEach(cleanup);

it("Header.js: renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Header />, div);
});

it("Header.js: matches snapshot", () => {
  const tree = testRenderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Header.js: renders 'Get assistance' button correctly", () => {
  const { getByTestId } = render(<Header />);
  expect(getByTestId("get-assistance")).toHaveTextContent(
    "Hafðu samband við þjónustuver bankans"
  );
});

it("Header.js: 'Get assistance' button leads to correct URL", () => {
  const { getByTestId } = render(<Header />);
  expect(getByTestId("get-assistance").getAttribute("href")).toBe(
    "https://www.islandsbanki.is/is/vara/thjonusta/hafa-samband"
  );
});

it("Header.js: facebook link opens correct URL", () => {
  const { getByTestId } = render(<Header />);
  expect(getByTestId("open-facebook").getAttribute("href")).toBe("https://www.messenger.com/t/iba.spjallmenni");
});

