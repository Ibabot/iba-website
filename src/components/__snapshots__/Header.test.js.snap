// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.js: matches snapshot 1`] = `
<header
  className="web-header"
  id="web-header"
>
  <div
    className="logo"
    data-aos="fade-right"
  >
    <img
      alt="Íslandsbankinn logo"
      className="logo"
      src="isb-logo.svg"
    />
  </div>
  <div
    className="web-header-text"
    data-aos="fade-left"
    data-aos-delay="100"
  >
    <h1>
      Segðu halló við Íba
    </h1>
    <h2
      className="bottom-space"
    >
      <strong>
        Íba er snjallmenni Íslandsbanka.
      </strong>
    </h2>
    <h2>
      Íba getur svarað almennum spurningum og fyrirspurnum hér í spjallinu.
    </h2>
    <a
      className="btn btn-red"
      data-testid="get-assistance"
      href="https://www.islandsbanki.is/is/vara/thjonusta/hafa-samband"
      rel="noopener noreferrer"
      target="_blank"
    >
      Hafðu samband við þjónustuver bankans
    </a>
    <p>
      Ég er líka á
       
      <a
        data-testid="open-facebook"
        href="https://www.messenger.com/t/iba.spjallmenni"
        rel="noopener noreferrer"
        target="_blank"
      >
        Facebook
      </a>
    </p>
  </div>
</header>
`;
